arbnode/dataposter/storage/storage.go

```diff
@@ -24,3 +28,125 @@ type QueuedTransaction struct {
 	Created         time.Time // may be earlier than the tx was given to the tx poster
 	NextReplacement time.Time
 }
+
+// LegacyQueuedTransaction is used for backwards compatibility.
+// Before https://github.com/OffchainLabs/nitro/pull/1773: the queuedTransaction
+// looked like this and was rlp encoded directly. After the pr, we are store
+// rlp encoding of Meta into queuedTransaction and rlp encoding it once more
+// to store it.
+type LegacyQueuedTransaction struct {
+	FullTx          *types.Transaction `rlp:"nil"`
```

Comment:
Why the `rlp:"nil"` here? I'm not sure if it's backwards compatible or not. It doesn't seem to have been in the old queued transaction struct. I think it might be for tests, in which case we should just fill in this value in tests (`types.NewTx` will make a tx from the Data field).

Reply:
Yes it was for tests, dropped it and adapted tests.